src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -3320,8 +3320,8 @@ void MacroAssembler::lookup_secondary_supers_table_slow_path(Register r_super_kl
   NearLabel L_huge;
 
   // The bitmap is full to bursting.
-  z_cghi(r_bitmap, Klass::SECONDARY_SUPERS_BITMAP_FULL);
-  z_bre(L_huge);
+  z_cghi(r_array_length, Klass::SECONDARY_SUPERS_BITMAP_FULL - 2);
```

Comment:
```suggestion
  z_chi(r_array_length, Klass::SECONDARY_SUPERS_BITMAP_FULL - 2);
```
This probably doesn't matter, but it's a 32-bit length.

Reply:
Correct, chi would be cleaner. cghi works too, because the length is loaded as 32 bit value. (The length is loaded as unsigned 32 bit with zero extend. Not sure if this is ideal, but negative length should not occur AFAIK.)

Reply:
That's what I was thinking. To use anything other than `chi` is confusing to the reader. (Well, it was confusing to this reader, anyway.)